include/inexor/vulkan-renderer/vkresult_to_text.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+namespace inexor::vulkan_renderer {
+
+/// @brief Turn a VkResult into a string.
+/// @note This function can be used for both VkResult error and success values.
+/// @param result The VkResult return value which will be turned into a string.
+[[nodiscard]] const char *get_vkresult_string(const VkResult result) {
```

Comment:
```suggestion
[[nodiscard]] std::string get_vkresult_string(const VkResult result) {
```

Reply:
@yeetari Didn't you prefer `const char*` for this?

Reply:
Yep, I can't see a reason to use `std::string`. It just means an extra allocation, which would be nice to avoid.

Reply:
Alternatively, I think `std::string_view` could be used which will do the same thing as `const char *`.

Reply:
> 
> 
> Yep, I can't see a reason to use `std::string`. It just means an extra allocation, which would be nice to avoid.

Oh right.

Also the function can be `constexpr` if iam not mistaken. Not sure if it is useful.

Reply:
"constexpr everything" -Jason Turner, haha

Reply:
Done.